### DISCUSSION TITLE
Are CmdNextTab and CmdPrevTab valid commands?

### DISCUSSION THREAD

**Discussion by WandersFar** [sumatrapdfreader/sumatrapdf#3226] [Q&A]
I saw them in the [source code:](https://github.com/sumatrapdfreader/sumatrapdf/blob/master/src/Commands.h#L9)

```
    V(CmdNextTab, "Next Tab")
    V(CmdPrevTab, "Previous Tab")
```

But not in the [Commands doc.](https://www.sumatrapdfreader.org/docs/Commands)

I tried to reference them in my config, but this doesn’t work:

```
	[
		Cmd = CmdPrevTab
		Key = Ctrl+PageUp
	]
	[
		Cmd = CmdNextTab
		Key = Ctrl+PageDown
	]
```

If CmdNextTab and CmdPrevTab are not valid, are there some other commands I can use in their place?

I would also like to assign Alt+1 through 9 to Ctrl+1 through 9, but I don’t see commands for that, either. Not even in the source code, unlike this pair.

Thanks for any help.

**Comment by WandersFar** [comment]
Okay, I just saw the other thread, and after updating to the pre-release version CmdPrevTab & CmdNextTab work.

So the only question I have is about Alt+1-9. Are there commands available to reassign these hotkeys?

**Comment by GitHubRulesOK** [comment]
hmmm
assigning tabs FROM current allocation of alternate numbers 1-8 and last (9) would have little appeal, 
however you can assign those key to other commands as per desire (if not wanted) so say close to the right (or any other CMD) can be alt +1 but then its just as easy to select with mouse right click the "tab context" rather than mouse and keys

**Comment by WandersFar** [comment]
I use Alt+Numbers as global media hotkeys, and I use Ctrl+Numbers in my browser fairly often, so I’m trying to replicate that behavior in Sumatra.

**Reply by GitHubRulesOK** [reply]
if you want to over-ride for global try
```
       [
		Cmd = CmdNone
		Key = Alt + 1

	]
```
etc.

**Comment by adiva** [comment]
I have tryed to add this shortcut:
```
Shortcuts [
	 [
		Cmd = CmdNextTab
		Key = Ctrl+ PgUp
	]
	 [
		Cmd = CmdNone
		Key = Ctrl+ PgDown
	]
]
```

into "**c:\Users\[myuser]\AppData\Local\SumatraPDF\SumatraPDF-settings.txt**"
but not works.

My shortcut are moved at the bottom of the config file, like this:

```
[...]
TimeOfLastUpdateCheck = 0 0
OpenCountWeek = 630

# Settings after this line have not been recognized by the current version
Shortcuts [
	 [
		Cmd = CmdNextTab
		Key = Ctrl+ PgUp
	]
	 [
		Cmd = CmdNone
		Key = Ctrl+ PgDown
	]
]
```

even using "SumatraPDF-3.4.6-64-install.exe"

**Reply by WandersFar** [reply]
You put spaces before PgUp and PgDown and you didn’t write out Page, that might be why. It should look like this:

	[
		Cmd = CmdPrevTab
		Key = Ctrl+PageUp
	]
	[
		Cmd = CmdNextTab
		Key = Ctrl+PageDown
	]

Also make sure you are running the latest beta release. It doesn’t work under the standard version.

**Reply by GitHubRulesOK** [reply]
I dont think spaces are the problem however the version that supported them would be Thus suprised that whole block should not be excluded as `shortcuts [ ]` were available in 3.4.6 including `Cmd = CmdNone` AHHhhhh as @WandersFar  poinnts out try `Ctrl + PageDown` but it needs be a 3.4.6 settings file.

![image](https://user-images.githubusercontent.com/12754913/215579752-b45b0286-8c11-463d-b6d0-dd1a62553b9d.png)

**Reply by adiva** [reply]
@WandersFar: 
spaces aren't the problem (for example "**Alt+Right**" or "**Alt + Right**" are the same), 
instead **PageUp** works, **PgUp** does't work.

@WandersFar / @GitHubRulesOK:
All working fine after installing [SumatraPDF pre-release](https://www.sumatrapdfreader.org/prerelease)

Now I use these cooool and speedy shortcuts (try them and let me know):
```
Shortcuts 
[
	[
		Cmd = CmdNextTab
		Key = Alt + Right
	]
	[
		Cmd = CmdPrevTab
		Key = Alt + Left
	]
	[
		Cmd = CmdCloseCurrentDocument
		Key = Alt + Down
	]
]

```
Thanks @WandersFar / @GitHubRulesOK

**Reply by kjk** [reply]
I've added `PgUp` and `PgDown` so they should be recognized in shortcuts.
